cloud_chatops/src/pr_reminder.py

```diff
@@ -0,0 +1,206 @@
+            self.prs, reminder_message, mention=False, private_user=channel
+        )
+
+    def post_reminder_message(self) -> WebClient.chat_postMessage:
+        """
+        This method posts the main reminder message to start the thread if PR notifications.
+        :return: The reminder message return object
+        """
+        reminder = self.client.chat_postMessage(
+            channel=self.CHANNEL,
+            text="Here are the outstanding PRs as of today:",
+        )
+        return reminder
+
+    def post_thread_messages(
+        self,
+        prs: Dict[str, List],
+        reminder_message: WebClient.chat_postMessage,
+        mention: bool,
+        private_user=None,
+    ) -> None:
+        """
+        This method collates each individual PR reminder message and calls a post method.
+        :param private_user: The user to filter messages for if chosen.
+        :param mention: To mention the users or not
+        :param prs: A list of PRs from GitHub
+        :param reminder_message: The reminder message object
+        """
+        private_count = 0
+        for repo in prs.values():
+            for pr in repo:
+                info = {
+                    "pr_title": f"{pr['title']} #{pr['number']}",
+                    "user": pr["user"]["login"],
+                    "url": pr["html_url"],
+                    "created_at": pr["created_at"],
+                    "channel": reminder_message.data["channel"],
+                    "thread_ts": reminder_message.data["ts"],
+                    "mention": mention,
+                    "draft": pr["draft"],
+                }
+                checked_info = self.check_pr(info)
+                if private_user:
+                    if info["user"] == private_user:
+                        private_count += 1
+                        self.send_thread(**checked_info)
+                    else:
+                        pass
+                else:
+                    self.send_thread(**checked_info)
+        if not private_count:
+            self.send_no_prs(reminder_message)
+
+    def send_no_prs(self, reminder: WebClient.chat_postMessage) -> None:
+        """
+        This method sends a message to the user that they have no PRs open.
+        This method is only called if no other PRs have been mentioned.
+        :param reminder: The thread message to send under.
+        """
+        self.client.chat_postMessage(
+            text="You have no outstanding pull requests open.",
+            channel=reminder.data["channel"],
+            thread_ts=reminder.data["ts"],
+            unfurl_links=False,
+        )
+
+    def check_pr(self, info: Dict) -> Dict:
+        """
+        This method validates certain information in the PR data such as who authored the PR and if it's old or not.
+        :param info: The information to validate.
+        :return: The validated information.
+        """
+        if info["user"] not in self.slack_ids:
```

Comment:
Which user is the default user?

Reply:
That would be David F, I'll add a comment